.github/workflows/release-shell-pkg.yaml

```diff
@@ -0,0 +1,48 @@
+name: Publish Shell Package
+
+# on:
+#   push:
+#     tags:
+#         - 'shell-pkg-v*'
+on:
+  pull_request_target:
```

Comment:
Ah, so the `pull_request_target` trigger provides access to secrets because it runs in the context of the target repo. That definitely makes sense.

Now that I'm thinking about this holistically, I'm wondering if pull requests are where we would want to publish shell releases. I would expect a release or a tag to trigger the publish to npm. Additionally, it appears that there are some security considerations to keep in mind when using this trigger

> Warning: For workflows that are triggered by the pull_request_target event, the GITHUB_TOKEN is granted read/write repository permission unless the permissions key is specified and the workflow can access secrets, even when it is triggered from a fork. Although the workflow runs in the context of the base of the pull request, you should make sure that you do not check out, build, or run untrusted code from the pull request with this event. [^1]

[^1]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

I originally left this comment in https://github.com/rancher/dashboard/pull/10103#discussion_r1426787558

Reply:
Yeah, the trigger is a tag like Rancher Components. I just wanna test it in a simpler way (on a PR). If it works, then I can finish (uncomment) the code on the workflow and just finalise the PR

Reply:
Kudos to Neil for discovering this 👏 👏 👏